### DISCUSSION TITLE
Migration plan for a lettuce or aloe implementation

### DISCUSSION THREAD

**Discussion by inoa-jboliveira** [behave/behave#1207] [Q&A] [answered]
I have an application with BDD tests originally written with [lettuce](https://pypi.org/project/lettuce) and then migrated to some awful fork called [aloe](https://pypi.org/project/aloe). Both libs are pretty much dead.

I see behave is the only still maintained cucumber implementation and it seems well accepted in the community.

The only reference I fond so far is this paragraph of data (notice the link for lettuce is no longer valid, you may not want to keep linking to it!):
https://behave.readthedocs.io/en/latest/comparison/#lettuce

Is there any definitive guide or scripts that help migrating from lettuce to Behave?
We have a django application, if that matters

**Comment by jenisys** [comment, marked as answer]
I am not aware of any tool that converts a `lettuce` project to a `behave` project.
Therefore, you probably need to do that by hand (at least initially).

* I assume that most of the work is in the conversion of the steps

DIFFERENCES:

* You should select an assert-matcher, like: [assertpy], [hamcrest] or …
* Step parameters: `behave` supports a number of step-matchers. `parse` is the default one, `cucumber-expressions` is the newest. You normal make a decision which one is used per project or per step-library (as default step-matcher). All except the regex step-matcher have type-conversion support for the parameters.
* Hooks: Are slightly different compared to `lettuce` (if I remember it correctly)
* …

[assertpy]: https://github.com/assertpy/assertpy
[hamcrest]: https://github.com/hamcrest/PyHamcrest

SEE ALSO:

* https://behave.readthedocs.io/en/latest/
* https://github.com/behave/behave.example repo (and its docs: https://behave.github.io/behave.example/ )
* Cucumber-expressions support (which covers some important aspects what you normally do in a new project): https://behave.readthedocs.io/en/latest/new_and_noteworthy_v1.2.7/#step-definitions-with-cucumber-expressions

NOTES:

* A comparison is pretty hard (by now) because the documentation website for `lettuce` is gone.
* The `lettuce` repo still exists: https://github.com/gabrielfalcao/lettuce

**Comment by jenisys** [comment]
Or ask ChatGPT or its equivalents, like:

* https://gurubase.io/g/behave/convert-lettuce-project-to-behave-bdd
* As always, the answer is not 100% correct. `behave` has also a `@step` decorator.
